vyos/vyos-1x
# dns: T3337: Add advanced options, dns zones

## Change Summary

The original author of the proposed code in the  task [T3337](https://phabricator.vyos.net/T3337)
I just create a PR.
That commit allows using advanced options for dns/zones.

## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T3337

## Component(s) name

dns
## Proposed changes


## How to test
Need to check with the author of the code.
For me, it is not clear which format we should use for PTR names.
```
set service dns forwarding advanced zone localhost.local a local.loc address '192.0.2.1'
set service dns forwarding advanced zone localhost.local mx mx5.local.loc hostname 'mx10-foo.local.loc'
set service dns forwarding advanced zone localhost.local ptr 4.2.0.192.in-addr.arpa. hostname '192.0.2.4'
set service dns forwarding advanced zone localhost.local ptr 2.2.0.192.in-addr.arpa. hostname '192.0.2.2'
set service dns forwarding advanced zone localhost.local ptr 8.2.0.192.in-addr.arpa hostname '192.0.2.8'
set service dns forwarding allow-from '100.64.0.0/24'
set service dns forwarding allow-from '192.0.2.0/24'
set service dns forwarding allow-from '192.168.122.0/24'
set service dns forwarding allow-from '127.0.0.1/32'
set service dns forwarding listen-address '100.64.0.1'

```

## Checklist:
